src/main/java/prefs/BoardPresetsLoader.java

```diff
@@ -0,0 +1,66 @@
+package prefs;
+
+import com.google.gson.*;
+import com.google.gson.reflect.TypeToken;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import util.HTLog;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BoardPresetsLoader {
+    private static final Logger logger = LogManager.getLogger(BoardPresetsLoader.class.getName());
+    private static final String CHARSET = "UTF-8";
+
+    private Gson gson;
+    private Map<String, List<PanelInfo>> boardPresets = new LinkedHashMap<>();
+
+    public BoardPresetsLoader() {
+        setupGson();
+        loadBoardPresets();
+    }
+
+    private void setupGson() {
+        gson = new GsonBuilder()
+                .setPrettyPrinting()
+                .registerTypeAdapter(LocalDateTime.class,
+                        (JsonSerializer<LocalDateTime>) (src, typeOfSrc, context) -> {
+                            Instant instant = src.atZone(ZoneId.systemDefault()).toInstant();
+                            long epochMilli = instant.toEpochMilli();
+                            return new JsonPrimitive(epochMilli);
+                        })
+                .registerTypeAdapter(LocalDateTime.class,
+                        (JsonDeserializer<LocalDateTime>) (json, typeOfT, context) -> {
+                            Instant instant = Instant.ofEpochMilli(json.getAsJsonPrimitive().getAsLong());
+                            return LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+                        }
```

Comment:
Are these necessary? LocalDateTime instances can be serialised by default, judging from similar code in Read/WriteTask.

If they are, they should be pulled out into an inner class.